### PR TITLE
Minor improvement for Helix bootstrap tool

### PR DESCRIPTION
**Pull request by jsjtzyy**
1. ensure there is a dry-run option for bootstrapping/upgrading Helix.
2. use isClusterSetup() instead of getClusters() to avoid expensive validation.

### CODE REVIEW

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 1]
```diff
@@ -863,22 +877,30 @@ private void addUpdateInstances(String dcName, Map<String, Set<String>> partitio
             // the appropriate instanceConfig that is constructed with the correct values from both).
             // For now, only bootstrapping cluster is allowed to directly change InstanceConfig
             dcAdmin.setInstanceConfig(clusterName, instanceName, instanceConfigFromStatic);
+          } else {
+            info(
```
Comment (xuhao417347761) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:881]: Minor: formating, info("[{}] Instance {} already present in Helix...",

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 2]
```diff
@@ -464,8 +465,21 @@ private HelixBootstrapUpgradeUtil(String hardwareLayoutPath, String partitionLay
         throw new IllegalArgumentException(
             entry.getKey() + " has invalid number of ZK endpoints: " + zkConnectStrs.size());
       }
-      HelixAdmin admin = helixAdminFactory.getHelixAdmin(zkConnectStrs.get(0));
+      HelixAdmin admin;
+      RealmAwareZkClient zkClient = null;
+      if (helixAdminFactory == null) {
+        RealmAwareZkClient.RealmAwareZkClientConfig clientConfig =
+            new RealmAwareZkClient.RealmAwareZkClientConfig().setConnectInitTimeout(30 * 1000L)
+                .setZkSerializer(new ZNRecordSerializer());
```
Comment (xuhao417347761) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:473]: The ZNRecordSerializer, HelixZkClient and ZKHelixAdmin are deprecated. Maybe add a to-do to update in the future?

Comment (jsjtzyy) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:473]: sure, will do

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java (hunk 1 above)
Comment (jsjtzyy) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:881]: Actually it has been formatted by intellij, as I did option+cmd+L.

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java [hunk 3]
```diff
@@ -145,7 +146,7 @@ public static void main(String[] args) {
   static void createCluster(String destZkString, String destClusterName) {
     HelixZkClient destZkClient = getHelixZkClient(destZkString);
     HelixAdmin destAdmin = new ZKHelixAdmin(destZkClient);
-    if (destAdmin.getClusters().contains(destClusterName)) {
+    if (ZKUtil.isClusterSetup(destClusterName, destZkClient)) {
```
Comment (jsjtzyy) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:149]: @ankagrawal this minor change should help us get rid of the warning messages we saw when building a new VCR cluster.

**Pull request merged**